vmware-research/certifier-framework-for-confidential-computing
# (#105,#112) Add CCA-shim support. Develop simple_app_under_cca using CCA-shim framework

This commit enhances the Certifier to work with CCA-shim support.
A new simple_app_under_cca is added to exercise this new platform.

Develop Certifier Framework and proofs support to work with CCA-shim:
[Co-authored by John Manferdelli]

- Add proofs logic to certlib_proofs.go
- Add VerifyCCAAttestation(), ValidateCCAEvidence()
- Update cca_shim.cc:cca_Attest(), cca_Verify() to fix attest and verify logic
- Changed cca_Attest to use hash of what_to_say
- Use hard-coded measurement of app to generate example_app.measurement
- Add temporary_instructions.txt to doc the workflow steps

Develop CCA-shim code and simple_app_under_cca example app:

- Add C-Go wrappers for Certifier to call into Islet cca_Verify() APIs
- Adds ccalib/cca_verify.cc, .h, cca_verify_dummy.cc, ccaverify/ccaverify.go

- Add simple_app_under_cca/ sample app
- Makefile support for shim_test, sample-app and Certifier artifacts
- Add support for new CCA-sample-app to run_example.sh
- Add tests to CI to:
- Download Islet SDK, make shim_test, attest_seal_test
- Build-and-test new sample app

Misc changes:
- run_example.sh now supports --no-make-clean which can be used to
  run 'compile_app', step and skip 'make clean'. This speeds up dev-cycles.
